modules/broker-amqp/src/main/java/org/wso2/broker/amqp/Server.java

```diff
@@ -69,23 +73,51 @@
     private EventExecutorGroup ioExecutors;
     private Channel plainServerChannel;
     private Channel sslServerChannel;
+    private boolean isHaEnabled = false;
+    private volatile boolean isStartCalled = false;
+    private volatile boolean isShutdownHookTriggered = false;
+    private AtomicBoolean isActive = new AtomicBoolean(false);
 
-    public Server(StartupContext context) throws Exception {
-        BrokerConfigProvider configProvider = context.getService(BrokerConfigProvider.class);
+    public Server(StartupContext startupContext) throws Exception {
+        BrokerConfigProvider configProvider = startupContext.getService(BrokerConfigProvider.class);
         this.configuration = configProvider
                 .getConfigurationObject(AmqpServerConfiguration.NAMESPACE, AmqpServerConfiguration.class);
-        this.broker = context.getService(Broker.class);
+        this.broker = startupContext.getService(Broker.class);
+        BrokerHaConfiguration.HaConfiguration brokerHaConfiguration =
+                configProvider.getConfigurationObject(
+                        BrokerHaConfiguration.NAMESPACE, BrokerHaConfiguration.HaConfiguration.class);
+        this.isHaEnabled = brokerHaConfiguration.isEnabled();
 
         if (broker == null) {
             throw new RuntimeException("Could not find the broker class to initialize AMQP server");
         }
-
         bossGroup = new NioEventLoopGroup();
         workerGroup = new NioEventLoopGroup();
         ioExecutors = new DefaultEventExecutorGroup(BLOCKING_TASK_EXECUTOR_THREADS);
+        HaStrategy haStrategy = startupContext.getService(HaStrategy.class);
+        if (isHaEnabled && haStrategy != null) {
+            haStrategy.registerListener(this, 2);
+        }
+
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            isShutdownHookTriggered = true;
+            try {
+                if (plainServerChannel != null) {
+                    plainServerChannel.close().sync();
+                }
+                if (sslServerChannel != null) {
+                    sslServerChannel.close().sync();
+                }
+            } catch (InterruptedException e) {
+                LOGGER.error("Error stopping transport ", e);
+            } finally {
+                shutdownExecutors();
+            }
+        }));
     }
 
     private void shutdownExecutors() {
+        LOGGER.info("Shutting Down Executors");
```

Comment:
Shall we be more specific? Maybe something similar to `Shutting Down AMQP Netty Executors`